### PR TITLE
Add support for linux-dmabuf surface feedback

### PR DESCRIPTION
**Pull request by emersion**
References: https://gitlab.freedesktop.org/wlroots/wlroots/-/merge_requests/1376

### CODE REVIEW

File: sway/server.c [hunk 1]
```diff
@@ -9,14 +9,17 @@
 #include <wlr/backend/multi.h>
 #include <wlr/backend/session.h>
 #include <wlr/config.h>
+#include <wlr/render/interface.h>
```
Comment (bl4ckb0ne) [inline sway/server.c:12]: I don't think this one is required

File: sway/tree/container.c [hunk 2]
```diff
@@ -1058,6 +1072,75 @@ static void set_fullscreen(struct sway_container *con, bool enable) {
...
+	}
+
+	if (!con->pending.workspace || !con->pending.workspace->output) {
+		return;
+	}
+
+	struct sway_output *output = con->pending.workspace->output;
+	struct wlr_output *wlr_output = output->wlr_output;
+
+	// TODO: add wlroots helpers for all of this stuff
```
Comment (bl4ckb0ne) [inline sway/tree/container.c:1092]: Would be nice to have an issue on wlroots side to track it

File: sway/tree/container.c [hunk 3]
```diff
@@ -1058,6 +1072,75 @@ static void set_fullscreen(struct sway_container *con, bool enable) {
...
+
+	const struct wlr_drm_format_set *renderer_formats =
+		wlr_renderer_get_dmabuf_texture_formats(server.renderer);
+	assert(renderer_formats);
+
+	int renderer_drm_fd = wlr_renderer_get_drm_fd(server.renderer);
+	int backend_drm_fd = wlr_backend_get_drm_fd(wlr_output->backend);
```
Comment (bl4ckb0ne) [inline sway/tree/container.c:1099]: Aren't those the same?

**Review by bl4ckb0ne — COMMENTED** [review]
Just a few questions, otherwise LGTM

File: sway/tree/container.c (hunk 2 above)
Comment (emersion) [reply sway/tree/container.c:1092]: Indeed, opened https://gitlab.freedesktop.org/wlroots/wlroots/-/issues/3343

File: sway/tree/container.c (hunk 3 above)
Comment (emersion) [reply sway/tree/container.c:1099]: Not necessarily:

- The renderer FD points to the DRM render node, the backend FD points to the primary node.
- On split render/display setups, backend FD points to the display-only device, renderer FD points to the render-only FD (well… not yet, but soon)
- When the user overrides the rendering device (`WLR_RENDER_DRM_DEVICE`), then they might not match.
- When we'll support multi-GPU scanout (https://gitlab.freedesktop.org/wlroots/wlroots/-/issues/3331), they might not match.

File: sway/server.c (hunk 1 above)
Comment (emersion) [reply sway/server.c]: Done!

File: sway/tree/container.c (hunk 3 above)
Comment (bl4ckb0ne) [reply sway/tree/container.c:1103]: Ah yes, the infamous split render/display setup, forgot about that one.

**Review by bl4ckb0ne — APPROVED** [review]
LGTM, thanks :)

**Pull request merged**